drivers/shared/executor/executor_linux.go

```diff
@@ -124,19 +124,18 @@ func (l *LibcontainerExecutor) ListProcesses() set.Collection[int] {
 	return procstats.List(l.command)
 }
 
-// cleanOldProcessesInCGroup kills processes that might ended up orphans when the
-// executor was unexpectedly killed and nomad can't reconnect to them.
-func (l *LibcontainerExecutor) cleanOldProcessesInCGroup(nomadRelativePath string) {
+// cleanOldProcessesInCGroup kills processes that might ended up orphans when
+// the executor was unexpectedly killed and nomad can't reconnect to them.
+func (l *LibcontainerExecutor) cleanOldProcessesInCGroup(nomadRelativePath string) error {
```

Comment:
Note to reviewers: I changed two things about this method. For one, it returns an error if it fails, because libcontainer 1.2+ will not launch a process in a cgroup that already has a process in it; not sure what exactly was the behavior before, but an error from this method can not be ignored. Secondly, new libcontainer is slightly faster in launching processes, so we end up having a race condition between the time the process gets a SIGKILL and the time its PID gets removed from the cgroups pidfile. Thus this method now only returns successfully once it makes sure the PID got removed.